### ISSUE TITLE
Longevity failure: missing output.log in vic-admin bundle

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#2872]
After almost 4 hours of runtime, hit the following issue:

```
    '-rw-r----- 0/0           65542 2016-10-24 05:13 docker/verbose
-rw-r----- 0/0           99683 2016-10-24 05:13 vpxd:vpxd.log
-rw-r----- 0/0         1608331 2016-10-24 05:13 vicadm/heap
-rw-r----- 0/0            1325 2016-10-24 05:13 lsmod
-rw-r----- 0/0          159458 2016-10-24 05:13 hostd/HostSystem:host-10 @ /stress-datacenter/host/cls/homer.eng.vmware.com
-rw-r----- 0/0          194449 2016-10-24 05:13 VCH-0-2745/vmware.log
-rw-r----- 0/0             996 2016-10-24 05:13 ip-addr
-rw-r----- 0/0              41 2016-10-24 05:13 vicadm/block
-rw-r----- 0/0             204 2016-10-24 05:13 free
-rw-r----- 0/0             129 2016-10-24 05:13 sys-block
-rw-r----- 0/0             265 2016-10-24 05:13 proc-self-fd
-rw-r----- 0/0          448810 2016-10-24 05:13 docker/heap
-rw-r----- 0/0              68 2016-10-24 05:13 disk-by-label
-rw-r----- 0/0          179702 2016-10-24 05:13 vic-init/heap
-rw-r----- 0/0         4398735 2016-10-24 05:13 portlayer/heap
-rw-r----- 0/0           77659 2016-10-24 05:13 vicadm/verbose
-rw-r----- 0/0            3624 2016-10-24 05:13 ps
-rw-r----- 0/0           21785 2016-10-24 05:13 mad_yalow-2f6aac4425b2734f7a6f5e520d4ff53e4d6eec2d523163321cdb4ea9835f4d5e/tether.debug
    [ Message content over the limit has been removed. ]
...--- 0/0         2463173 2016-10-24 05:13 var/log/vic/docker-personality.log
-rw-r----- 0/0        13957528 2016-10-24 05:13 var/log/vic/init.log
-rw-r----- 0/0            1537 2016-10-24 05:13 proc-mounts
-rw-r----- 0/0             519 2016-10-24 05:13 ip-link
-rw-r----- 0/0              41 2016-10-24 05:13 portlayer/block
-rw-r----- 0/0         3512627 2016-10-24 05:13 var/log/vic/port-layer.log
-rw-r----- 0/0            7734 2016-10-24 05:13 netstat
-rw-r----- 0/0            8462 2016-10-24 05:13 vicadm/concise
-rw-r----- 0/0          262130 2016-10-24 05:13 hostd/HostSystem:host-17 @ /stress-datacenter/host/cls/nelson.eng.vmware.com
-rw-r----- 0/0           21651 2016-10-24 05:13 portlayer/verbose
-rw-r----- 0/0             381 2016-10-24 05:13 df
-rw-r----- 0/0            5029 2016-10-24 05:13 vic-init/concise
-rw-r----- 0/0           14378 2016-10-24 05:13 portlayer/concise
-rw-r----- 0/0          213235 2016-10-24 05:13 hostd/HostSystem:host-19 @ /stress-datacenter/host/cls/milhouse.eng.vmware.com
-rw-r----- 0/0              41 2016-10-24 05:13 docker/block
-rw-r----- 0/0         3646174 2016-10-24 05:13 var/log/vic/vicadmin.log' does not contain '2f6aac4425b2734f7a6f5e520d4ff53e4d6eec2d523163321cdb4ea9835f4d5e/output.log'
```

The final bundle that was collected actually has the output.log in it, so maybe some sort of delay/race in grabbing the file by vic-admin?

[14-1-Longevity-VCH-0-2745-container-logs.zip](https://github.com/vmware/vic/files/548994/14-1-Longevity-VCH-0-2745-container-logs.zip)

**Comment by mhagen-vmware** [comment]
I do see this issue in vicadmin.log:
```
time="2016-10-24T10:13:50Z" level=warning msg="error reading mad_yalow-2f6aac4425b2734f7a6f5e520d4ff53e4d6eec2d523163321cdb4ea9835f4d5e/output.log({%!s(*object.Datastore=&{{/stress-datacenter/datastore/vsanDatastore 0xc420a8cf00 {Datastore datastore-252}}}) e2de0d58-2bcb-ac02-8faa-1866da12435b/output.log}): 404 Not Found\n"
```

**Comment by mhagen-vmware** [comment]
There is the same error for vmware.log as well, and both appear to be missing from the bundle whereas the tether.log doesn't have the error and is the only log of the three that is present in the bundle.

**Comment by mhagen-vmware** [comment]
```
time="2016-10-24T10:13:50Z" level=warning msg="error reading mad_yalow-2f6aac4425b2734f7a6f5e520d4ff53e4d6eec2d523163321cdb4ea9835f4d5e/vmware.log({%!s(*object.Datastore=&{{/stress-datacenter/datastore/vsanDatastore 0xc420a8cf00 {Datastore datastore-252}}}) e2de0d58-2bcb-ac02-8faa-1866da12435b/vmware.log}): 404 Not Found\n"
```

**Comment by sgairo** [comment]
Does the final bundle collected end up having vmware.log as well?

**Comment by mhagen-vmware** [comment]
yes

**Comment by sgairo** [comment]
It looks like only vmware.log is collected in final bundle for applianceVM whereas all three (output,tether,vmware) are collected per containerVM in the final bundle. Is this expected or should all three be shown with applianceVM as well?

**Comment by mhagen-vmware** [comment]
Saw this again on the latest longevity run after about 12 hours.
Container name is gigantic_woznia-95ed4106dc47e64d9d7c71dcb954f910a3da852b4467321542dfa19bdb873d3c

[container-logs.zip](https://github.com/vmware/vic/files/597840/container-logs.zip)

**Comment by hmahmood** [comment]
Is this a dup of https://github.com/vmware/vic/issues/2827? I looked at the regression test, and we do a create, start, ps, and then stop in quick succession. It could be that the output of the file never makes it to the datastore (output.log) the same way that it does not make it to the terminal. 

@caglar10ur has recently merged in https://github.com/vmware/vic/pull/3185; that very likely would have fixed any truncation issues.

**Comment by mhagen-vmware** [comment]
Moving out of GA for now.  We believe that if you retry, the log will show up eventually.  Evemtually need to re-address this code and make it cleaner.

**Comment by sgairo** [comment]
Okay, talked with Hassan about it today and have an approach, thanks Mike!

**Comment by jzt** [comment]
This happened on the 14th (new record!) iteration of the longevity tests:

```
'-rw-r----- 0/0              63 2016-11-22 02:07 uptime
-rw-r----- 0/0            3428 2016-11-22 02:07 ps
-rw-r----- 0/0            1195 2016-11-22 02:07 meminfo
-rw-r----- 0/0             997 2016-11-22 02:07 ip-addr
-rw-r----- 0/0          757891 2016-11-22 02:07 journalctl
-rw-r----- 0/0           14377 2016-11-22 02:07 portlayer/concise
-rw-r----- 0/0              41 2016-11-22 02:07 vicadm/block
-rw-r----- 0/0             178 2016-11-22 02:07 ip-route
-rw-r----- 0/0           20504 2016-11-22 02:07 vicadm/verbose
-rw-r----- 0/0         1621489 2016-11-22 02:07 var/log/vic/docker-personality.log
-rw-r----- 0/0            1209 2016-11-22 02:07 lsmod
-rw-r----- 0/0            9109 2016-11-22 02:07 vicadm/concise
-rw-r----- 0/0           11380 2016-11-22 02:07 vic-init/heap
-rw-r----- 0/0              41 2016-11-22 02:07 docker/block
-rw-r----- 0/0         2092666 2016-11-22 02:07 var/log/vic/port-layer.log
-rw-r----- 0/0          227942 2016-11-22 02:07 hostd/HostSystem:host-17 @ /stress-datacenter/host/cls/nelson.eng.vmware.com
-rw-r----- 0/0              41 2016-11-22 02:08 vic-init/block
-rw-r----- 0/0             266 2016-11-22 02:08 proc-self-fd
    [ Message content over the limit has been removed. ]
-rw-r----- 0/0              41 2016-11-22 02:08 portlayer/block
-rw-r----- 0/0          179599 2016-11-22 02:08 pensive_brattai-1175b24a12fc141efda72f9ab62b5a30e64d230db95622607dcd3f24dbcef67f/vmware.log
-rw-r----- 0/0            1711 2016-11-22 02:08 proc-mounts
-rw-r----- 0/0             381 2016-11-22 02:08 df
-rw-r----- 0/0             517 2016-11-22 02:08 ip-link
-rw-r----- 0/0              82 2016-11-22 02:08 disk-by-path
-rw-r----- 0/0          857964 2016-11-22 02:08 vicadm/heap
-rw-r----- 0/0          300306 2016-11-22 02:08 docker/heap
-rw-r----- 0/0          710553 2016-11-22 02:08 var/log/vic/vicadmin.log
-rw-r----- 0/0           93208 2016-11-22 02:08 vpxd:vpxd.log
-rw-r----- 0/0          210407 2016-11-22 02:08 hostd/HostSystem:host-15 @ /stress-datacenter/host/cls/ralph.eng.vmware.com
-rw-r----- 0/0            6214 2016-11-22 02:08 netstat
-rw-r----- 0/0             113 2016-11-22 02:08 lsblk
-rw-r----- 0/0             129 2016-11-22 02:08 sys-block
-rw-r----- 0/0           21684 2016-11-22 02:08 portlayer/verbose
-rw-r----- 0/0          258849 2016-11-22 02:08 hostd/HostSystem:host-19 @ /stress-datacenter/host/cls/milhouse.eng.vmware.com' does not contain '1175b24a12fc141efda72f9ab62b5a30e64d230db95622607dcd3f24dbcef67f/tether.debug'
```

VCH Name:  VCH-0-3171
VC: bart.eng.vmware.com
VCH:  10.17.109.115 (it's still up, just idling)

[container-logs.zip](https://github.com/vmware/vic/files/607276/container-logs.zip)

**Comment by hmahmood** [comment]
This is a different error though; it is complaining about tether.debug missing, not output.log.

**Comment by sgairo** [comment]
Could it be a result of the same problem that caused both errors?

**Comment by sgairo** [comment]
Output.log ended up showing in the bundle when it was al said and done also

**Comment by hickeng** [comment]
#3247 introduces a work around for this problem (retry + delay) but does not fix it.

**Comment by hmahmood** [comment]
I am wondering if this may be just vsan flakiness. The file seems to appear and disappear, as evidenced by its existence in the log bundle. Both tether.debug and output.log are tied to serial ports. Other files in the log bundle don't seem to be affected, which would suggest vsan is _not_ to blame.

**Comment by sgairo** [comment]
@hmahmood With some time to marinate, are you still thinking it could be vsan? Or a mixture of vsan and something else?

**Comment by hmahmood** [comment]
@sgairo have we confirmed this _only_ happens when vsan is in the mix?

**Comment by sgairo** [comment]
I believe I merged the fix for this in #4019, will confirm.

**Comment by mhagen-vmware** [comment]
It does appear to be fixed on the latest longevity run, as longevity ran for almost 60 hrs before hitting a different issue.  This issue was being hit, much sooner than 60 hrs in the past, so I am going to close this for now, but we can re-open if we see it again.
